libraries/AP_Mission/AP_Mission.cpp

```diff
@@ -45,6 +52,12 @@ void AP_Mission::init()
     if (sizeof(union Content) != 12) {
         AP_HAL::panic("AP_Mission Content must be 12 bytes");
     }
+    
+    // If Mission retain bit is not set then it should clear the mission
+    if(!is_option(MASK_MISSION_RETAIN)) {
```

Comment:
`!is_option()` doesn't seem like it's actually cleaner then just doing `mask & MISSION_MASK_CLEAR` (making the bit set to flag clear on reboot would also simplify the code at runtime without any user comprehension problems).

Reply:
I agree, I don't see the need for the method and the MASK should be CLEAR.